lib/SIL/IR/SILFunction.cpp

```diff
@@ -16,6 +16,7 @@
 #include "swift/AST/Availability.h"
 #include "swift/AST/Expr.h"
 #include "swift/AST/GenericEnvironment.h"
+#include "swift/AST/IRGenOptions.h"
```

Comment:
I don't think it makes sense for files in lib/SIL to depend on IRGen headers. Can you find another way to plumb this through?

Reply:
Since this already merged in main I opened https://github.com/apple/swift/pull/73727 to address this issue

Reply:
I cherry-picked https://github.com/apple/swift/pull/73727 here as well